### PR TITLE
fix(mmu): invalidCodePrefix with empty memory

### LINKED ISSUE #1309: Index out of bond in MMU instruction  `INVALID_CODE_PREFIX`
ref test 121

### PR DESCRIPTION
**Pull request by letypequividelespoubelles**

### CODE REVIEW

File: arithmetization/src/main/java/net/consensys/linea/zktracer/module/mmu/instructions/InvalidCodePrefix.java [hunk 1]
```diff
@@ -77,14 +76,12 @@ public MmuData preProcess(MmuData mmuData) {
...
-        Bytes.of(
-            mmuData
-                .sourceRamBytes()
-                .get((int) (LLARGE * initialSourceLimbOffset + initialSourceByteOffset)));
-    Bytes arg1 = microLimb;
-    Bytes arg2 = Bytes.of(EIP_3541_MARKER);
-    boolean result = wcp.callEQ(arg1, arg2);
+    final int offset = (int) (LLARGE * initialSourceLimbOffset + initialSourceByteOffset);
+    final boolean offsetOutOfBond = offset >= mmuData.sourceRamBytes().size();
+    microLimb = offsetOutOfBond ? Bytes.of(0) : Bytes.of(mmuData.sourceRamBytes().get(offset));
```
Comment (letypequividelespoubelles) [inline arithmetization/src/main/java/net/consensys/linea/zktracer/module/mmu/instructions/InvalidCodePrefix.java]: The only thing that matters, the rest is cosmetic

Comment (OlivierBBB) [reply arithmetization/src/main/java/net/consensys/linea/zktracer/module/mmu/instructions/InvalidCodePrefix.java:80]: Do we not have direct access to the value of `offset` without recomputing it as `LLARGE * initialSourceLimbOffset + initialSourceByteOffset` ?

Comment (OlivierBBB) [reply arithmetization/src/main/java/net/consensys/linea/zktracer/module/mmu/instructions/InvalidCodePrefix.java:80]: `offsetOutOfBond` should be `offsetIsOutOfBounds`

Comment (OlivierBBB) [reply arithmetization/src/main/java/net/consensys/linea/zktracer/module/mmu/instructions/InvalidCodePrefix.java]: Do we not have direct access to the value of `offset` without recomputing it as `LLARGE * initialSourceLimbOffset + initialSourceByteOffset` ?

Comment (OlivierBBB) [reply arithmetization/src/main/java/net/consensys/linea/zktracer/module/mmu/instructions/InvalidCodePrefix.java]: `offsetOutOfBond` should be `offsetIsOutOfBounds`

**Review by OlivierBBB — APPROVED** [review]
LGTM, only minor comments

File: arithmetization/src/main/java/net/consensys/linea/zktracer/module/mmu/instructions/InvalidCodePrefix.java (hunk 1 above)
Comment (letypequividelespoubelles) [reply arithmetization/src/main/java/net/consensys/linea/zktracer/module/mmu/instructions/InvalidCodePrefix.java]: Yes & Yes

**Pull request merged**